pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -934,7 +934,41 @@ func ValidateVirtualMachineInstanceSpec(field *k8sfield.Path, spec *v1.VirtualMa
 			Field:   field.Child("GPUs").String(),
 		})
 	}
-
+	if spec.Domain.Devices.HostDevices != nil && !config.HostDevicesPassthroughEnabled() {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("Host Devices feature gate is not enabled in kubevirt-config"),
+			Field:   field.Child("HostDevices").String(),
+		})
+	}
+	if hostDevs := config.GetPermittedHostDevices(); hostDevs != nil {
+		// build a map of all permitted host devices
+		supportedHostDevicesMap := make(map[string]bool)
+		for _, dev := range hostDevs.PciHostDevices {
+			supportedHostDevicesMap[dev.ResourceName] = true
+		}
+		for _, dev := range hostDevs.MediatedDevices {
```

Comment:
So we smash all PCI and mediated devices into the same map? I assume names can't ever overlap?

Reply:
Yes. Normally, pci / mdev resources are named differently to reflect the model of the assigned device.